src/components.tsx

```diff
@@ -239,11 +239,12 @@ export function A(props: AnchorProps) {
       state={JSON.stringify(props.state)}
       classList={{
         ...(props.class && { [props.class]: true }),
-        [props.inactiveClass!]: !isActive(),
-        [props.activeClass!]: isActive(),
+        [props.inactiveClass!]: !isActive()[0],
+        [props.activeClass!]: isActive()[0] && !isActive()[1],
```

Comment:
Change to 
```
[props.activeClass!]: isActive()[0] && (!props.exactActiveClass || !isActive()[1]),
```
Would allow for loose matching if no `exactActiveClass` was provided.